diff -up hdf-4.2.9/hdf/test/sdmms.c.wrap hdf-4.2.9/hdf/test/sdmms.c
--- hdf-4.2.9/hdf/test/sdmms.c.wrap	2013-02-07 21:44:52.000000000 -0700
+++ hdf-4.2.9/hdf/test/sdmms.c	2013-02-16 20:20:35.741870559 -0700
@@ -63,9 +63,10 @@ static int32 ical5;
 void
 test_sdmms(void)
 {
-    int         i, j, err, err1, err2, ret;
-    intn        rank;
-    int32       dims[2];
+    unsigned int i, j;
+    int          err, err1, err2, ret;
+    intn         rank;
+    int32        dims[2];
 
     rank = 2;
     dims[0] = 10;
@@ -87,7 +88,7 @@ test_sdmms(void)
                 ui16[i][j] = (uint16) ((i * 6000) + j);     /* range: 0 ~ 60000 */
 
                 i32[i][j] = (int32) ((i * 60000) - 300000 + j);     /* range: 0 ~ 2-billion */
-                ui32[i][j] = (uint32) ((i * 400000000) + j);    /* range: 0 ~ 4-billion */
+                ui32[i][j] = (uint32) ((i * 400000000U) + j);    /* range: 0 ~ 4-billion */
             }
 
           f32scale[i] = (float32) ((i * 40) + j);   /* range: 0 ~ 4-billion */
@@ -99,7 +100,7 @@ test_sdmms(void)
           ui16scale[i] = (uint16) ((i * 6000) + j);     /* range: 0 ~ 60000 */
 
           i32scale[i] = (int32) ((i * 60000) - 300000 + j);     /* range: 0 ~ 2-billion */
-          ui32scale[i] = (uint32) ((i * 400000000) + j);    /* range: 0 ~ 4-billion */
+          ui32scale[i] = (uint32) ((i * 400000000U) + j);    /* range: 0 ~ 4-billion */
       }
 
     cal1 = (float64) 10.0;
