--- setup.py~	2013-10-22 12:51:49.000000000 +0100
+++ setup.py	2013-10-22 12:56:34.000000000 +0100
@@ -555,6 +555,7 @@
         if sys.version_info[0] > 2:
             cmake_cmd.append("-DPYTHON3_EXECUTABLE=%s" % self.py_executable)
             cmake_cmd.append("-DPYTHON3_INCLUDE_DIR=%s" % self.py_include_dir)
+            cmake_cmd.append("-DPYTHON3_INCLUDE_PATH=%s" % self.py_include_dir)
             cmake_cmd.append("-DPYTHON3_LIBRARY=%s" % self.py_library)
             if self.build_type.lower() == 'debug':
                 cmake_cmd.append("-DPYTHON3_DBG_EXECUTABLE=%s" % self.py_executable)
@@ -562,6 +563,7 @@
         else:
             cmake_cmd.append("-DPYTHON_EXECUTABLE=%s" % self.py_executable)
             cmake_cmd.append("-DPYTHON_INCLUDE_DIR=%s" % self.py_include_dir)
+            cmake_cmd.append("-DPYTHON_INCLUDE_PATH=%s" % self.py_include_dir)
             cmake_cmd.append("-DPYTHON_LIBRARY=%s" % self.py_library)
             if self.build_type.lower() == 'debug':
                 cmake_cmd.append("-DPYTHON_DEBUG_LIBRARY=%s" % self.py_library)
